theano/gpuarray/dnn.py

```diff
@@ -1860,6 +1860,8 @@ def local_gpua_pool_dnn_alternative(op, ctx_name, inputs, outputs):
     if not op.ignore_border:
         return
     img, = inputs
+    if len(op.ds) not in (2, 3) or img.ndim != len(op.ds) + 2:
```

Comment:
If the second condition is false, there is a much bigger problem. The graph isn't consistent. I'm fine to keep that condition there, but we should catch that earlier. I think it is done elsewhere.